crates/ide/src/highlight_related.rs

```diff
@@ -285,11 +290,129 @@ fn highlight_references(
     if res.is_empty() { None } else { Some(res.into_iter().collect()) }
 }
 
+pub(crate) fn highlight_branches(
```

Comment:
why is hte implementation of this so different from the test. I'd expect this to look fairly similar to `hl_exit_points`